apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -293,7 +294,9 @@ start_link(MgrId, ResId, Group, ResourceType, Config, Opts) ->
         config = Config,
         opts = Opts,
         state = undefined,
-        error = undefined
+        %% `undefined` error may signify error state with undefined reason,
+        %% `none` explicitly means that no error occurred
+        error = none
```

Comment:
Nit. Having a term of type `{error, _Reason :: term()} | undefined` here should unambiguous? Right now AFAICS it's ambiguous with respect to `{error, none}` error from a healthcheck.

Reply:
Thanks, fixed.